src/pages/studyView/StudyViewPage.tsx

```diff
@@ -160,22 +163,23 @@ export default class StudyViewPage extends React.Component<
         delete (window as any).studyPageFilter;
 
         if (hashString) {
-            hashString = hashString.substring(1);
-            hashString.split('&').forEach(datum => {
-                if (datum.startsWith('filterJson')) {
-                    const filters = datum.match(/filterJson=([^&]*)/);
-                    if (filters && filters.length > 1) {
-                        newStudyViewFilter.filters = filters[1];
-                    }
-                } else if (datum.startsWith('sharedGroups')) {
-                    const sharedGroups = datum.match(/sharedGroups=([^&]*)/);
-                    if (sharedGroups && sharedGroups.length > 1) {
-                        newStudyViewFilter.sharedGroups = sharedGroups[1];
-                        // Open group comparison manager if there are shared groups in the url
-                        this.store.showComparisonGroupUI = true;
-                    }
-                }
-            });
+            const params = parse(hashString);
```

Comment:
Do you think it makes sense to type the hash object?

Reply:
+1 - `parse(hashString) as something`

Reply:
updated